x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning_logic.ts

```diff
@@ -155,4 +248,268 @@ export const RelevanceTuningLogic = kea<
+      const { http } = HttpLogic.values;
+      const { search_fields: searchFields, boosts } = removeBoostStateProps(values.searchSettings);
+      const url = `/api/app_search/engines/${engineName}/search_settings_search`;
+
+      actions.setResultsLoading(true);
+
+      try {
+        const response = await http.post(url, {
+          query: {
+            query,
+          },
+          body: JSON.stringify({
+            boosts: isEmpty(boosts) ? undefined : boosts,
+            search_fields: isEmpty(searchFields) ? undefined : searchFields,
+          }),
+        });
+
+        actions.setSearchResults(response.results);
+      } catch (e) {
+        flashAPIErrors(e);
+      }
+    },
+    setSearchSettings: () => {
+      actions.getSearchResults();
+    },
+    onSearchSettingsSuccess: ({ searchSettings }) => {
+      actions.setSearchSettingsResponse(searchSettings);
+      actions.getSearchResults();
+      window.scrollTo(0, 0);
+    },
+    onSearchSettingsError: () => {
+      window.scrollTo(0, 0);
+    },
+    updateSearchSettings: async () => {
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      const url = `/api/app_search/engines/${engineName}/search_settings`;
+
+      try {
+        const response = await http.put(url, {
+          body: JSON.stringify(removeBoostStateProps(values.searchSettings)),
+        });
+        setSuccessMessage(UPDATE_SUCCESS_MESSAGE);
+        actions.onSearchSettingsSuccess(response);
+      } catch (e) {
+        flashAPIErrors(e);
+        actions.onSearchSettingsError();
+      }
+    },
+    resetSearchSettings: async () => {
+      if (window.confirm(RESET_CONFIRMATION_MESSAGE)) {
+        const { http } = HttpLogic.values;
+        const { engineName } = EngineLogic.values;
+
+        const url = `/api/app_search/engines/${engineName}/search_settings/reset`;
+
+        try {
+          const response = await http.post(url);
+          setSuccessMessage(DELETE_SUCCESS_MESSAGE);
+          actions.onSearchSettingsSuccess(response);
+        } catch (e) {
+          flashAPIErrors(e);
+          actions.onSearchSettingsError();
+        }
+      }
+    },
+    toggleSearchField: ({ name, disableField }) => {
+      const { searchSettings } = values;
+
+      const searchFields = disableField
+        ? omit(searchSettings.search_fields, name)
+        : { ...searchSettings.search_fields, [name]: { weight: 1 } };
+
+      actions.setSearchSettings({
+        ...searchSettings,
+        search_fields: searchFields,
+      });
+    },
+    updateFieldWeight: ({ name, weight }) => {
+      const { searchSettings } = values;
+      const { search_fields: searchFields } = searchSettings;
+
+      actions.setSearchSettings({
+        ...searchSettings,
+        search_fields: {
+          ...searchFields,
+          [name]: {
+            ...searchFields[name],
+            weight: parseFloat(weight.toFixed(1)),
```

Comment:
Sorry, I think our discussion about whether we allow numbers vs strings of numbers has me confused. Shouldn't we do `parseFloat` first and then `toFixed` in case a `'2'` is being passed?

```suggestion
            weight: parseFloat(weight).toFixed(1),
```

Reply:
I ended up reverting this change. `toFixed` returns a string which is why I have to call parseFloat. The string representation may actually not be parseable back to a float in certain cases. I think the previous implementation with Math.round is the appropriate solution.

Reply:
Weight is actually a number, just for reference, it can never be a string.

Reply:
Ohh gotcha gotcha, super interesting! If we're doing this in multiple places it _might_ be nice to pull it out to a util so we can clearly explain in the fn comments why we're doing it this way, etc., and so we don't forget and used toFixed later on in a different place